src/DynamoCoreWpf/Views/Search/LibraryView.xaml.cs

```diff
@@ -180,19 +184,48 @@ private void ExpandCategory(TreeViewItem sender, NodeCategoryViewModel selectedC
             foreach (var categoryToBeCollapsed in allExpandedCategories)
             {
                 // If class button was clicked.
-                if (categoryToBeExpanded == null)
+                if (selectedClass.SubCategories.Count == 0)
                 {
                     var categoryClasses = categoryToBeCollapsed.Items[0] as ClassesNodeCategoryViewModel;
```

Comment:
I'm pretty sure I have asked this before, but forgive me I'm going to ask again: are we sure that `ClassesNodeCategoryViewModel` will always be at index 0 for all scenarios?